DEHPCommon/UserPreferenceHandler/UserPreferenceService/UserPreferenceService.cs

```diff
@@ -0,0 +1,177 @@
+                {
+                    var serializer = new JsonSerializer();
+                    var result = (List<ServerConnection>)serializer.Deserialize(file, typeof(List<ServerConnection>));
+                    this.userPreferenceSettings.SavedServerConections.AddRange(result);
+                    return (T)this.userPreferenceSettings;
+                }
+            }
+            catch (IOException ex)
+            {
+                logger.Error(ex, "IOException- The user preference could not be found");
+
+                throw new Exception("The user preference could not be found", ex);
+            }
+            catch (Exception ex)
+            {
+                logger.Error(ex, "The user preference could not be read");
+
+                throw new Exception("The user preference could not be read", ex);
+            }
+        }
+
+        /// <summary>
+        /// Writes the <see cref="UserPreference"/> to disk
+        /// </summary>
+        /// <param name="userPreference">
+        /// The <see cref="UserPreference"/> that will be persisted
+        /// </param>
+        public void Write<T>(T userPreference) where T : UserPreference
+        {
+            if (userPreference == null)
+            {
+                throw new ArgumentNullException(nameof(userPreference), "This may not be null");
+            }
+
+            var assemblyName = this.QueryAssemblyTitle(userPreference.GetType());
+
+            this.CheckConfigurationDirectory();
+
+            var path = Path.Combine(this.ApplicationDirectory, $"{assemblyName}{SETTING_FILE_EXTENSION}");
+
+            logger.Debug("Write user preference to for {0} to {1}", assemblyName, path);
+
+            using (var streamWriter = File.CreateText(path))
+            {
+                var serializer = new JsonSerializer
+                {
+                    ContractResolver = new CamelCasePropertyNamesContractResolver(),
+                    Formatting = Formatting.Indented
+                };
+
+                serializer.Serialize(streamWriter, userPreference.SavedServerConections);
+            }
+
+            this.userPreferenceSettings = userPreference;
+        }
+
+        /// <summary>
+        /// Queries the name of the assembly that contains the <see cref="Type"/>
+        /// </summary>
+        /// <param name="type">
+        /// The <see cref="Type"/> that is contained in the assembly for which the name is queried.
+        /// </param>
+        /// <returns>
+        /// A string that contains the name of the assembly
+        /// </returns>
+        private string QueryAssemblyTitle(Type type)
+        {
+            return ((AssemblyTitleAttribute)type.Assembly.GetCustomAttribute(typeof(AssemblyTitleAttribute)))
+                .Title;
+        }
+
+        /// <summary>
+        /// Configuration file Directory
+        /// </summary>
+        public string ApplicationDirectory
```

Comment:
The name of this property does not reflect what it is. also please put properties before methods.